### PR TITLE
Edit type constraints docs for style and flow

### PR DESCRIPTION
**Pull request by laurapacilio**
This PR concludes the optional attributes experiment, and revises the associated documentation for type constraints. It attempts to make the writing more concise and also use conventions like bullets to help make the content clearer.

It also:
- Adds a note at the top of the section that this functionality is only available in v.1.3 and later (following the convention we've used for this type of notice in other places in our docs).
- Removes a section about how this is experimental. My understanding is that the goal is to end the experiment when 1.3 is launched, and thus this section will not be necessary. Please let me know if I'm mistaken!

**ADD REDIRECT:** When we remove pages, we need to add a redirect. I created a companion PR in `terraform-website` that adds a redirect for the deleted optional function page: : https://github.com/hashicorp/terraform-website/pull/2342
I will wait to merge redirect PR until v1.3 is launched!

------------

Screen Shot 2022-06-08 at 12 45 18 PM

### CODE REVIEW

**Comment by laurapacilio** [comment]
@alisdair no rush! Also, please push back where I've misinterpreted the meaning or used the wrong terms for programming concepts. I tried to stay consistent, but I am also not a developer :)

**Comment by laurapacilio** [comment]
@alisdair thank you for looking at this! I think it makes sense to coordinate docs and code, so please push changes to this branch as you see fit! That way, we can just launch everything at once, which I think makes sense. I'm assuming the plan is for all of this to go out for 1.3.

And then once you do that, let me know if/when it's okay merge things in. I can also leave that to you if that's easier.
Thank you!

**Comment by laurapacilio** [comment]
nice!! :-) Thanks, Alisdair!

File: website/docs/language/expressions/type-constraints.mdx [hunk 1]
```diff
@@ -276,29 +276,19 @@ variable "with_optional_attribute" {
...
 ### Example: Nested Structures with Optional Attributes and Defaults
 
-The following configuration defines a variable which describes a number of storage buckets, each of which is used to host a website. This variable type uses several optional attributes, one of which is itself an `object` type with optional attributes and defaults.
+The following example defines a variable for storage buckets that host a website. This variable type uses several optional attributes, including `website`, which is itself an optional `object` type that has optional attributes and defaults.
 
 ```hcl
 terraform {
-  # Optional attributes are currently experimental.
   experiments = [module_variable_optional_attrs]
 }
```
Comment (apparentlymart) [inline website/docs/language/expressions/type-constraints.mdx:293]: This entire block should be removed once the experiment is ended; `experiments = [module_variable_optional_attrs]` will start returning an error once the experiment is concluded.

Comment (alisdair) [reply website/docs/language/expressions/type-constraints.mdx]: Thanks, fixed!

**Review by alisdair — APPROVED** [review]
This should be okay to merge now! We've received enough user feedback that I'm now confident that the updated `optional` design will go into 1.3 with the experiment concluded.

**Pull request merged**